File: lift/src/main/scala/jp_zipcode_search/model/Log.scala
package jp_zipcode_search.model

import java.util._
import scala.collection.jcl.Conversions._
import net.liftweb.util._
import com.google.appengine.api.datastore._
import Query._
import SortDirection._
import FetchOptions.Builder._

class Log(entity:Entity) {
  
  def this() = this(new Entity("Log"))
  
  def date = {
    try {
      entity.getProperty("date").asInstanceOf[java.util.Date]
    } catch {
      case e:NullPointerException => ""
    }
  }
    
}

object Log {
  
  val ds:DatastoreService = DatastoreServiceFactory.getDatastoreService
  
  def create(entity:Entity) = new Log(entity)
  
  def updatedAt = {
    var query = new Query("Log")
    query.addSort("date", DESCENDING)
    var res = ds.prepare(query).asList(withOffset(0).limit(1))
    if (res.length > 0) {
      val f = new Formatter
      f.format("%1$ta, %1$te %1$tb %1$tY %1$tH:%1$tM:%1$tS %1$tz", res.toList.map(create)(0).date).toString
    } else {
      ""
    }
  }
  
}

File: slim3/src/main/java/jp_zipcode_search/model/Log.java
package jp_zipcode_search.model;

import java.util.Date;
import com.google.appengine.api.datastore.Key;
import org.slim3.datastore.Attribute;
import org.slim3.datastore.Model;

@Model
public class Log {

  @Attribute(primaryKey = true)
  private Key key;

  private Date date = new Date();

  public Key getKey() {
    return key;
  }

  public void setKey(Key key) {
    this.key = key;
  }

  public Date getDate() {
    return date;
  }

  public void setDate(Date date) {
    this.date = date;
  }

}
